lib/block_view_atomic_txns.go

```diff
@@ -0,0 +1,474 @@
+	// on the blockchain in the order specified or none of the transactions
+	// will be executed.
+	//
+	// The AtomicTxnsMetadata.Txns field must be a specially formed
+	// slice of DeSo transactions to ensure their atomic execution on the blockchain.
+	// If this field is not properly structured, the AtomicTxns 'wrapper' transaction
+	// will be rejected. The transactions in AtomicTxnsMetadata.Txns and their corresponding
+	// ExtraData must form a circular doubly linked list. The links are embedded in the extra data map as follows:
+	// ** Take special note of the encoding schema for the AtomicTxnsChainLength **
+	//
+	// For the first transaction:
+	// AtomicTxnsMetadata.Txns[0].ExtraData = {
+	// 		AtomicTxnsChainLength: 		UintToBuf(uint64(len(AtomicTxnsMetadata.Txns)))...
+	// 		NextAtomicTxnPreHash:  		AtomicTxnsMetadata.Txns[1].AtomicHash()
+	//		PreviousAtomicTxnPreHash: 	AtomicTxnsMetadata.Txns[len(AtomicTxnsMetadata.Txns)-1].AtomicHash()
+	// }
+	//
+	// For the ith transaction where 0 < i < len(AtomicTxnsMetadata.Txns)-1:
+	// AtomicTxnsMetadata.Txns[i].ExtraData = {
+	// 		NextAtomicTxnPreHash:  		AtomicTxnsMetadata.Txns[i+1].AtomicHash()
+	//		PreviousAtomicTxnPreHash: 	AtomicTxnsMetadata.Txns[i-1].AtomicHash()
+	// }
+	//
+	// For the last transaction:
+	// AtomicTxnsMetadata.Txns[len(AtomicTxnsMetadata.Txns)-1].ExtraData = {
+	// 		NextAtomicTxnPreHash:  		AtomicTxnsMetadata.Txns[0].AtomicHash()
+	//		PreviousAtomicTxnPreHash: 	AtomicTxnsMetadata.Txns[len(AtomicTxnsMetadata.Txns)-2].AtomicHash()
+	// }
+	//
+	// The "AtomicHash()" function is a special transaction hash taken without consideration for the signature
+	// on a transaction as well as certain extra data fields. Otherwise, constructing an atomic transaction
+	// would be impossible as deriving the links using MsgDeSoTxn.Hash() would have circular dependencies.
+	// The purpose of using the AtomicHash for links is to prevent a malicious 3rd party from injecting or
+	// modifying the transactions included in the atomic transaction. This helps ensure the atomicity of the
+	// atomic transactions. NOTE: The MsgDeSoTxn.AtomicHash() operation DOES keep the AtomicTxnsChainLength
+	// key in the ExtraData map to ensure that start of the chain is not compromised.
+	//
+	// The AtomicTxnsChainLength key is crucial for pinning the start of the atomic transaction. It's
+	// arbitrary and redundant that we use the chains length, but it adds an extra sanity check when
+	// connecting the transaction to the blockchain. Without a key representing the starting transaction,
+	// a malicious entity could reorder the transactions while still preserving the validity of the hashes
+	// in the circularly linked list. The AtomicTxnsChainLength included in the first transaction ensures
+	// the transactions are atomically executed in the order specified.
+	// NOTE: As a measure to reduce potential mempool divergences, only one transaction can have an
+	// AtomicTxnsChainLength key.
```

Comment:
This is an excellent comment. Great job codifying all the logic into an easy-to-understand comment.

Reply:
So did we get rid of the ability to do an "unordered" list of txns? I think it's probably good to kill it for now since it could open up potential attack vectors.